internal/pkg/crawl/link_header.go

```diff
@@ -0,0 +1,52 @@
+package crawl
+
+import (
+	"regexp"
+	"strings"
+)
+
+// Represents a Link struct, containing a URL to which it links, and a Rel to define the relation
+type Link struct {
+	URL string
+	Rel string
+}
+
+// Parse parses a raw Link header in the form:
+//
+//	<url1>; rel="what", <url2>; rel="any"; another="yes", <url3>; rel="thing"
+//
+// returning a slice of Link structs
+// Each of these are separated by a `, ` and the in turn by a `; `, with the first always being the url, and the remaining the key-val pairs
+// See: https://simon-frey.com/blog/link-header/, https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link
+func Parse(link string) []Link {
```

Comment:
Because regular expressions can be CPU intensive, can you implement the `Parse` function without them? One way to do this would be using the Go strings `Split` and `Trim` functions to breakdown the input string by the separator characters defined in the [Link Header specification](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link), which are `;` and `,` and `<>`. You already started this approach using `Split` on `,` so great job there!

Reply:
Should I completely ignore malformed inputs, or just try to parse url out of it?
e.g.

 - `<>;` this can completely be ignored
 
But url can be extracted from these

 - `https://test.com>; rel=;`
 - `https://another.com;;;;`

Reply:
If you can try and succeed at parsing URLs out of malformed inputs, no harm in trying to. Just make sure you are not parsing false positives 99% of the time. (1% of false positives, _just making up numbers here_, is fine..)

Reply:
Okay! Done